### PR TITLE
CSPL:1451: Backport of CSPL-1414, improve logging of App Framework tests

### PR DESCRIPTION
**Pull request by jambrosiano**

### CODE REVIEW

File: test/s1/appframework/appframework_test.go [hunk 1]
```diff
@@ -338,80 +342,76 @@ var _ = Describe("s1appfw test", func() {
...
 
-			// Verify App Downlaod State on CR
-			// testenv.VerifyAppListPhaseStandalone(deployment, testenvInstance, deployment.GetName(), appSourceName, enterpriseApi.PhaseDownload, appFileList)
-
-			// Verify Apps are downloaded on Splunk Operator Pod
-			// testenvInstance.Log.Info("Verify Apps are downloaded on Splunk Operator container for apps", "version", appVersion)
-			// testenv.VerifyAppsDownloadedOnContainer(deployment, testenvInstance, testenvInstance.GetName(), []string{opPod}, appFileList, opLocalAppPathStandalone)
-			// Wait for Standalone to be in UPDATING status
-			testenv.VerifyStandalonePhase(deployment, testenvInstance, deployment.GetName(), splcommon.PhasePending)
-
```
Comment (pdhanoyasplunk) [inline test/s1/appframework/appframework_test.go:396]: Re Add back to the test.

File: test/s1/appframework/appframework_test.go [hunk 2]
```diff
@@ -425,25 +425,32 @@ var _ = Describe("s1appfw test", func() {
...
+			appVersion := "V1"
+			testenvInstance.Log.Info(fmt.Sprintf("Upload %s apps to S3 for Standalone and Monitoring Console", appVersion))
 			s3TestDirMC := "s1appfw-mc-" + testenv.RandomDNSName(4)
 			appFileList := testenv.GetAppFileListPhase3(appListV1)
 			uploadedFiles, err := testenv.UploadFilesToS3(testS3Bucket, s3TestDirMC, appFileList, downloadDirV1)
-			Expect(err).To(Succeed(), "Unable to upload apps to S3 test directory")
+			Expect(err).To(Succeed(), fmt.Sprintf("Unable to upload %s apps to S3 test directory for Monitoring Console", appVersion))
+			uploadedApps = append(uploadedApps, uploadedFiles...)
+			s3TestDir = "s1appfw-" + testenv.RandomDNSName(4)
+			uploadedFiles, err = testenv.UploadFilesToS3(testS3Bucket, s3TestDir, appFileList, downloadDirV1)
```
Comment (pdhanoyasplunk) [inline test/s1/appframework/appframework_test.go:452]: Add comment for upload of apps to Standalone.

**Review by pdhanoyasplunk — CHANGES_REQUESTED** [review]

File: test/testenv/verificationutils.go [hunk 3]
```diff
@@ -684,7 +683,7 @@ func VerifyAppsInFolder(deployment *Deployment, testenvInstance *TestEnv, ns str
 		for _, app := range apps {
 			folderName := app + "/"
 			found := CheckStringInSlice(appList, folderName)
-			logf.Log.Info("Check App in Directory", "Pod Name", podName, "App-name", folderName, "Path", path, "status", found)
+			logf.Log.Info("App check", "On pod", podName, "check app", folderName, "is in path", path, "Status", found)
 			if found != checkAppDirectory {
```
Comment (smohan-splunk) [inline test/testenv/verificationutils.go:687]: Do we need similar log updated in `VerifyAppsDownloadedOnContainer` ?

**Review by pdhanoyasplunk — APPROVED** [review]

File: test/testenv/verificationutils.go (hunk 3 above)
Comment (pdhanoyasplunk) [reply test/testenv/verificationutils.go:687]: I will handle that update as part of my change.

File: test/m4/appframework/appframework_test.go [hunk 4]
```diff
@@ -138,184 +143,185 @@ var _ = Describe("m4appfw test", func() {
...
 
-			// Verify apps are copied to location for M4 SVA
-			testenvInstance.Log.Info("Verify apps are copied to correct location based on Pod KIND for app", "version", appVersion)
+			// Verify V2 apps are copied to location for M4 SVA
+			testenvInstance.Log.Info(fmt.Sprintf("Verify %s apps are copied to correct location on M4 pods", appVersion))
 			testenv.VerifyAppsCopied(deployment, testenvInstance, testenvInstance.GetName(), allPodNames, appListV2, true, true)
 
-			// Verify apps are copied to location for Monitoring Console
-			testenvInstance.Log.Info("Verify apps are copied to correct location for Monitoring Console", "version", appVersion)
+			// Verify V2 apps are copied to location for Monitoring Console
```
Comment (smohan-splunk) [inline test/m4/appframework/appframework_test.go:275]: We don't call these as "V2" apps in the comments of the original PR. Should develop branch be updated to be consistent?

File: test/m4/appframework/appframework_test.go [hunk 5]
```diff
@@ -366,174 +379,187 @@ var _ = Describe("m4appfw test", func() {
...
+			// Verify Monitoring Console is Ready and stays in ready state
 			testenv.VerifyMonitoringConsoleReady(deployment, deployment.GetName(), mc, testenvInstance)
 
-			// Verify apps are downloaded by init-container on CM, Deployer
+			//########## INITIAL VERIFICATIONS ##########
+			// Verify V2 apps are downloaded on Cluster Manager and Deployer
 			initContDownloadLocation := "/init-apps/" + appSourceName
 			podNames := []string{fmt.Sprintf(testenv.ClusterManagerPod, deployment.GetName()), fmt.Sprintf(testenv.DeployerPod, deployment.GetName())}
-			testenvInstance.Log.Info("Verify apps are downloaded by init container for apps", "version", appVersion)
+			testenvInstance.Log.Info(fmt.Sprintf("Verify %s apps are downloaded on Cluster Manager and Deployer", appVersion))
```
Comment (smohan-splunk) [inline test/m4/appframework/appframework_test.go:411]: Log not matching on develop
`testenvInstance.Log.Info(fmt.Sprintf("Verify %s apps are downloaded for M4", appVersion))`

File: test/m4/appframework/appframework_test.go [hunk 6]
```diff
@@ -366,174 +379,187 @@ var _ = Describe("m4appfw test", func() {
...
 			testenv.VerifyAppsDownloadedOnContainer(deployment, testenvInstance, testenvInstance.GetName(), podNames, appFileList, initContDownloadLocation)
 
-			// Verify apps are downloaded by init-container on Monitoring Console
+			// Verify V2 apps are downloaded on Monitoring Console
 			initContDownloadLocationMCPod := "/init-apps/" + appSourceNameMC
 			mcPodName := fmt.Sprintf(testenv.MonitoringConsolePod, mcName, 0)
-			testenvInstance.Log.Info("Verify apps are downloaded by init container on MC pod for apps", "POD", mcPodName, "version", appVersion)
+			testenvInstance.Log.Info(fmt.Sprintf("Verify %s apps are downloaded on Monitoring Console POD %s", appVersion, mcPodName))
```
Comment (smohan-splunk) [inline test/m4/appframework/appframework_test.go:417]: Log not matching on develop
`testenvInstance.Log.Info(fmt.Sprintf("Verify %s apps are downloaded for Monitoring Console POD %s", appVersion, mcPodName))`

File: test/m4/appframework/appframework_test.go [hunk 7]
```diff
@@ -366,174 +379,187 @@ var _ = Describe("m4appfw test", func() {
...
+			testenvInstance.Log.Info(fmt.Sprintf("Verify %s apps are copied to correct location for Monitoring Console", appVersion))
 			testenv.VerifyAppsCopied(deployment, testenvInstance, testenvInstance.GetName(), []string{mcPodName}, appListV2, true, false)
 
-			// Verify apps are not copied in /etc/apps/ on CM and on Deployer (therefore not installed on Deployer and on CM)
-			testenvInstance.Log.Info("Verify apps are NOT copied to /etc/apps on CM and Deployer for app", "version", appVersion, "App List", appFileList)
+			// Verify V2 apps are not copied in /etc/apps/ on Cluster Manager and on Deployer (therefore not installed on Deployer and on Cluster Manager)
+			testenvInstance.Log.Info(fmt.Sprintf("Verify %s apps are NOT copied to /etc/apps on Cluster Manager and Deployer (App list: %s)", appVersion, appFileList))
 			testenv.VerifyAppsCopied(deployment, testenvInstance, testenvInstance.GetName(), podNames, appListV2, false, false)
 
-			// Verify apps are installed on (cluster-wide)
```
Comment (smohan-splunk) [inline test/m4/appframework/appframework_test.go:429]: Comment not matching on develop

File: test/m4/appframework/appframework_test.go [hunk 8]
```diff
@@ -366,174 +379,187 @@ var _ = Describe("m4appfw test", func() {
...
-			testenvInstance.Log.Info("Verify newer version of apps are installed on the pods", "version", appVersion)
+			// Verify V2 apps are installed on (cluster-wide)
+			testenvInstance.Log.Info(fmt.Sprintf("Verify %s apps are installed on the pods", appVersion))
 			testenv.VerifyAppInstalled(deployment, testenvInstance, testenvInstance.GetName(), allPodNames, appListV2, true, "enabled", true, true)
 
-			// Verify apps are installed on MC (local install)
-			testenvInstance.Log.Info("Verify apps are installed on the pods by running Splunk CLI commands for app", "version", appVersion)
+			// Verify V2 apps are installed on Monitoring Console (local install)
```
Comment (smohan-splunk) [inline test/m4/appframework/appframework_test.go:449]: Comment not matching

File: test/c3/appframework/appframework_test.go [hunk 9]
```diff
@@ -128,177 +137,196 @@ var _ = Describe("c3appfw test", func() {
...
+			testenvInstance.Log.Info(fmt.Sprintf("Verify %s apps are installed on Monitoring Console", appVersion))
 			testenv.VerifyAppInstalled(deployment, testenvInstance, testenvInstance.GetName(), []string{mcPodName}, appListV1, true, "enabled", false, false)
 
+			//############### UPGRADE APPS ################
 			// Delete apps on S3
-			testenvInstance.Log.Info("Delete apps on S3 for", "Version", appVersion)
+			testenvInstance.Log.Info(fmt.Sprintf("Delete %s apps on S3", appVersion))
 			testenv.DeleteFilesOnS3(testS3Bucket, uploadedApps)
 			uploadedApps = nil
 			testenvInstance.Log.Info("Testing upgrade scenario")
```
Comment (smohan-splunk) [inline test/c3/appframework/appframework_test.go:215]: Observation: Log not present in develop

File: test/c3/appframework/appframework_test.go [hunk 10]
```diff
@@ -339,179 +369,196 @@ var _ = Describe("c3appfw test", func() {
...
 			   * Prepare and deploy C3 CRD with app config and wait for pods to be ready
-			   ################## VERIFICATIONS #############
+			   ########## INITIAL VERIFICATIONS ############
 			   * Verify bundle push is successful
-			   * Verify apps are copied, installed on SH,Indexers pods
-			   ##########  SCALING UP ###########
-			   * Scale up indexers and Search Heads
+			   * Verify apps are copied, installed on Search Heads and Indexers
+			   #############  SCALING UP ###################
+			   * Scale up Indexers and Search Heads
```
Comment (smohan-splunk) [inline test/c3/appframework/appframework_test.go:534]: In develop, case difference
`* Scale up indexers and Search Heads`

File: test/c3/appframework/appframework_test.go [hunk 11]
```diff
@@ -1043,102 +1113,111 @@ var _ = Describe("c3appfw test", func() {
...
 			*/
 
+			//################## SETUP ####################
 			// Split Applist into 2 lists for local and cluster install
+			appVersion := "V2"
 			appListLocal := appListV2[len(appListV2)/2:]
 			appListCluster := appListV2[:len(appListV2)/2]
 
-			// Upload appListLocal to S3
+			// Upload appListLocal list of apps to S3 (to be used for local install)
```
Comment (smohan-splunk) [inline test/c3/appframework/appframework_test.go:1207]: Comment different in develop

File: test/m4/appframework/appframework_test.go (hunk 4 above)
Comment (jambrosiano) [reply test/m4/appframework/appframework_test.go:275]: I would say it's still ok, the appVersion is present in the "testenvInstance.Log.Info" lines of that code block

**Pull request merged**